### ISSUE TITLE
Learning rate remains 0 if batch_size is set to 1

### ISSUE THREAD

**Issue by yuh-yang** [Lightning-AI/lit-gpt#779]
Thanks for the brilliant work! I find something seems to be a bug at the fine-tuning process. But setting `batch_size=1`, which causes `is_accumulating` to be always true. Then, `step_count` can not be updated because only when `is_accumulating=False` can it be added 1:

```
        if not is_accumulating:
            optimizer.step()
            optimizer.zero_grad()
            if step_count > warmup_steps:
                scheduler.step()
            step_count += 1

```

Then, lr remains 0. during the whole training, because:

```
        if step_count <= warmup_steps:
            # linear warmup
            lr = learning_rate * step_count / warmup_steps
            for param_group in optimizer.param_groups:
```
